### PR TITLE
Add a check if Internet is available

### PR DESCRIPTION
**Pull request by badlee**
*OfflineBuilder* can now check if internet is avaible by lookup google (or other host), if is not avaible the connection is sate to *ConnectivityResult.none*

I have add two properties to **OfflineBuilder**
```dart
bool checkHost; // allow perform a DNS request to check if internet is avaible, the default value false
String hostToCheck; // Hostname to check, default value is google.com
```

Usage 
```dart
import 'package:flutter/material.dart';
import 'package:flutter_offline/flutter_offline.dart';

class DemoPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text("Offline Demo"),
      ),
      body: OfflineBuilder(
        hostTocheck : "google.ca", // perform a DNS request to check if internet is avaible
        checkHost : true, // the default falue is false, 
        connectivityBuilder: (
          BuildContext context,
          ConnectivityResult connectivity,
          Widget child,
        ) {
          final bool connected = connectivity != ConnectivityResult.none;
          return new Stack(
            fit: StackFit.expand,
            children: [
              Positioned(
                height: 24.0,
                left: 0.0,
                right: 0.0,
                child: Container(
                  color: connected ? Color(0xFF00EE44) : Color(0xFFEE4400),
                  child: Center(
                    child: Text("${connected ? 'ONLINE' : 'OFFLINE'}"),
                  ),
                ),
              ),
              Center(
                child: new Text(
                  'Yay!',
                ),
              ),
            ],
          );
        },
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            new Text(
              'There are no bottons to push :)',
            ),
            new Text(
              'Just turn off your internet.',
            ),
          ],
        ),
      ),
    );
  }
}
```

### CODE REVIEW

**Comment by jogboms** [comment]
@badlee this is a good addition. I would do a review ASAP.

**Comment by jogboms** [comment]
@badlee this took an awfully long time than expected. I have been swamped with normal work. Could you maybe update the tests for this?

**Comment by badlee** [comment]
@jogboms yes I'll do it.

File: lib/src/main.dart [hunk 1]
```diff
@@ -86,7 +101,11 @@ class OfflineBuilderState extends State<OfflineBuilder> {
...
         );
       },
-    ).transform(
+    );
+    _connectivityStream = widget.checkHost ? _connectivityStream.transform(
+       checkIfHostIsAvailble(widget.hostToCheck),
+    ) : _connectivityStream;
+     _connectivityStream = _connectivityStream.transform(
       debounce(widget.debounceDuration),
     );
```
Comment (jogboms) [inline lib/src/main.dart]: How about..

```
if (widget.checkHost) {
  _connectivityStream = _connectivityStream.transform(
     checkIfHostIsAvailble(widget.hostToCheck),
  );
}

_connectivityStream = _connectivityStream.transform(
  debounce(widget.debounceDuration),
);
```

**Comment by JamesMcIntosh** [comment]
Hi @jogboms @badlee
Just wondering if you are going to fix and push this PR through to master as it's been 3 months?
Many thanks

**Comment by jogboms** [comment]
@badlee do you wish to continue on this, or i should i take it over?

**Pull request closed without merging**